api/src/modules/indicators/indicator.repository.ts

```diff
@@ -1,10 +1,37 @@
-import { DataSource, Repository } from 'typeorm';
-import { Indicator } from 'modules/indicators/indicator.entity';
-import { Injectable } from '@nestjs/common';
+import { DataSource, In, Repository } from 'typeorm';
+import {
+  Indicator,
+  INDICATOR_STATUS,
+} from 'modules/indicators/indicator.entity';
+import { BadRequestException, Injectable } from '@nestjs/common';
 
 @Injectable()
 export class IndicatorRepository extends Repository<Indicator> {
   constructor(private dataSource: DataSource) {
     super(Indicator, dataSource.createEntityManager());
   }
+
+  async checkActiveIndicatorsForCalculations(
+    indicatorIds: string[],
+  ): Promise<void> {
+    const inactiveSelectedIndicators: Indicator[] = await this.dataSource
+      .getRepository(Indicator)
+      .find({
+        where: {
+          id: In(indicatorIds),
+          status: INDICATOR_STATUS.INACTIVE,
+        },
+      });
+
+    if (inactiveSelectedIndicators.length) {
+      const inactiveIndicatorsNames: string[] = inactiveSelectedIndicators.map(
+        (indicator: Indicator) => indicator.name,
+      );
+      throw new BadRequestException(
+        `Requested Indicators are not activated: ${inactiveIndicatorsNames.join(
+          ', ',
+        )}`,
+      );
+    }
+  }
```

Comment:
Since this validation is more relevant to a higher level in the chain (at controller level), it could maybe make more sense to move this logic up to the service method. This way you can use the regular findAll() with the required criteria, and no new method on the repository is needed.

Reply:
thanks, makes more sense! changes applied